pkg/defaults/revision_defaults.go

```diff
@@ -0,0 +1,50 @@
+package defaults
+
+import (
+	"context"
+
+	"knative.dev/pkg/apis"
+
+	"knative.dev/serving/pkg/apis/serving"
+	v1 "knative.dev/serving/pkg/apis/serving/v1"
+)
+
+const (
+	// GroupName is the group name for istio labels and annotations
+	GroupName = "service.istio.io"
+
+	// RevisionLabelKey is the label key attached to k8s resources to indicate
+	// which Revision triggered their creation.
+	RevisionLabelKey = GroupName + "/canonical-revision"
+
+	// ServiceLabelKey is the label key attached to a Route and Configuration indicating by
+	// which Service they are created.
+	ServiceLabelKey = GroupName + "/canonical-service"
+)
+
+// IstioRevision is a wrapper around Revision for setting Istio specific defaults
+type IstioRevision struct {
+	v1.Revision `json:",inline"`
+}
+
+// Verify that Revision adheres to the appropriate interfaces.
+var (
+	// Check that Revision can be validated, can be defaulted, and has immutable fields.
+	_ apis.Validatable = (*IstioRevision)(nil)
+	_ apis.Defaultable = (*IstioRevision)(nil)
+
+	// // Check that Revision can be converted to higher versions.
+	// _ apis.Convertible = (*Revision)(nil)
+
+	// // Check that we can create OwnerReferences to a Revision.
+	// _ kmeta.OwnerRefable = (*Revision)(nil)
+)
+
+// SetDefaults implements apis.Defaultable
+func (r *IstioRevision) SetDefaults(ctx context.Context) {
+	if r.Labels == nil {
+		r.Labels = make(map[string]string)
+	}
+	r.Labels[RevisionLabelKey] = r.Name
+	r.Labels[ServiceLabelKey] = r.Labels[serving.ServiceLabelKey]
```

Comment:
Note a revision will always have a parent configuration but it may not have a parent knative service.

https://github.com/knative/serving/pull/7081/files#diff-aa0a40159a3087a142da90ae81d95862R41-R63

Reply:
Ah thanks for the suggestion! Added.